File: api-common/src/main/java/a/gleb/apicommon/clientmanager/model/AccountRequestModel.java
/*
 * Copyright (c) 12/28/22, 7:57 PM.
 * Created by https://github.com/alwayswanna
 *
 */

package a.gleb.apicommon.clientmanager.model;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.*;

import java.time.LocalDate;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Schema(description = "Смена данных учетной записи.")
public class AccountRequestModel {

    @Schema(description = "Имя пользователя", example = "username")
    private String username;

    @Schema(description = "Пароль пользователя", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
    private String password;

    @Schema(description = "Электронный адрес", example = "olegIv@mail.ru")
    private String email;

    @Schema(description = "Имя", example = "Олег")
    private String firstName;

    @Schema(description = "Фамилия", example = "Иванов")
    private String middleName;

    @Schema(description = "Отчество", example = "Игоревич")
    private String lastName;

    @Schema(description = "Тип аккаунта", example = "COMPANY")
    private AccountType accountType;

    @Schema(description = "Дата рождения", example = "22.03.1995")
    private LocalDate birthDate;
}

File: fellowworkerfront/lib/models/account_request_model.dart
/*
 * Copyright (c) 1-2/8/23, 11:05 PM
 * Created by https://github.com/alwayswanna
 */

import 'package:json_annotation/json_annotation.dart';

part '../generated/account_request_model.g.dart';

@JsonSerializable()
class AccountRequestModel {
  final String? username;
  final String? password;
  final String? email;
  final String? firstName;
  final String? middleName;
  final String? lastName;
  final String? accountType;
  final String? birthDate;

  const AccountRequestModel(
      {required this.username,
      required this.password,
      required this.email,
      required this.firstName,
      required this.middleName,
      required this.lastName,
      required this.accountType,
      required this.birthDate});

  factory AccountRequestModel.fromJson(Map<String, dynamic> json) => _$AccountRequestModelFromJson(json);

  Map<String, dynamic> toJson() => _$AccountRequestModelToJson(this);
}
